Remotion/Development/UnitTesting.IsolatedCodeRunner.UnitTests/IsolatedCodeRunnerTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Configuration;
+using System.IO;
+using NUnit.Framework;
+
+namespace Remotion.Development.UnitTesting.IsolatedCodeRunner.UnitTests
+{
+  [TestFixture]
+  public class IsolatedCodeRunnerTests
+  {
+    private static bool _shouldStayFalse;
+
+    [Test]
+    public void Run_ExecutesCodeInNewProcess ()
+    {
+      var appProcessRunner = new IsolatedCodeRunner(TestAction);
+      appProcessRunner.Run();
+
+      Assert.That(appProcessRunner.Output.Trim(), Is.EqualTo("ran"));
+      Assert.That(_shouldStayFalse, Is.EqualTo(false));
+
+      static void TestAction (string[] args)
+      {
+        Console.WriteLine("ran");
+        _shouldStayFalse = true;
+      }
+    }
+
+    [Test]
+    public void Run_ExceptionIsRethrownInThisProcess ()
+    {
+      var appProcessRunner = new IsolatedCodeRunner(TestAction);
+      Assert.That(
+          () => appProcessRunner.Run(),
+          Throws.InvalidOperationException
+              .With.Message.EqualTo("This is bad."));
+
+      static void TestAction (string[] args)
+      {
+        Console.WriteLine("HEHEA");
+        throw new InvalidOperationException("This is bad.");
+      }
+    }
+
+    [Test]
+    public void Run_ArgumentsArePassedToNewProcess ()
+    {
+      var appProcessRunner = new IsolatedCodeRunner(TestAction);
+      appProcessRunner.Run("a", "b", "c");
+
+      static void TestAction (string[] args)
+      {
+        Assert.That(args, Is.EquivalentTo(new[] { "a", "b", "c" }));
```

Comment:
```suggestion
        Assert.That(args, Is.EqualTo(new[] { "a", "b", "c" }));
```
The order of items is important.